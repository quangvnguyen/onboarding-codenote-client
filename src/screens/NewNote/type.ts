export interface INewNoteProps {
  history: {
    push: (url: string) => void;
  };
}

export interface INewNoteStates {
  isLoading: boolean;
  content: string;
}