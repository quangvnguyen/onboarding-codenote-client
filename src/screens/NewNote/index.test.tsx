import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';
import NewNote from './index';
import config from '../../config';

let store;
const fileMock = {
  size: 0,
};
describe('New Note component', () => {
  const mockStore = configureMockStore();
  beforeAll(() => {
    store = mockStore({
      authenticate: {}
    });
  })

  it('renders without crashing', () => {
    const wrapper = shallow(
      <NewNote store={store} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders when submit with file bigger than MAX_ATTACHMENT_SIZE', () => {
    const wrapper = mount(
      <NewNote store={store} />
    );

    wrapper.instance().file = {
      size: config.MAX_ATTACHMENT_SIZE + 1,
    }

    const component = wrapper.find('form');
    component.simulate('submit');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders when submit', () => {
    const wrapper = mount(
      <NewNote store={store} />
    );

    const component = wrapper.find('form');
    component.simulate('submit');
    expect(wrapper).toMatchSnapshot();
  });

  it('should handleChange function called', () => {
    const event = {target: {id: 'content', value: 'content'}};
    const wrapper = mount(
      <NewNote store={store} />
    );

    wrapper.find('#content').simulate('change', event);
    expect(wrapper.state().content).toEqual('content');
  });

  it('should handleFileChange function called', () => {
    const event = {target: {id: 'file', files: [fileMock]}};
    const wrapper = mount(
      <NewNote store={store} />
    );

    wrapper.find('#file').simulate('change', event);
    expect(wrapper.instance().file).toEqual(fileMock);
  });
})