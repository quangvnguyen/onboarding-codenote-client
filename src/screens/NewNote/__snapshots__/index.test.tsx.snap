// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Note component renders when submit 1`] = `
<NewNote
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="NewNote"
  >
    <form
      onSubmit={[Function]}
    >
      <FormGroup
        bsClass="form-group"
        controlId="content"
      >
        <div
          className="form-group"
        >
          <FormControl
            bsClass="form-control"
            componentClass="textarea"
            onChange={[Function]}
            value=""
          >
            <textarea
              className="form-control"
              id="content"
              onChange={[Function]}
              value=""
            />
          </FormControl>
        </div>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        controlId="file"
      >
        <div
          className="form-group"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="file"
            >
              Attachment
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="file"
          >
            <input
              className=""
              id="file"
              onChange={[Function]}
              type="file"
            />
          </FormControl>
        </div>
      </FormGroup>
      <LoaderButton
        block={true}
        bsSize="large"
        bsStyle="primary"
        disabled={true}
        isLoading={true}
        loadingText="Creating…"
        text="Create"
        type="submit"
      >
        <Button
          active={false}
          block={true}
          bsClass="btn"
          bsSize="large"
          bsStyle="primary"
          className="LoaderButton "
          disabled={true}
          type="submit"
        >
          <button
            className="LoaderButton  btn btn-lg btn-primary btn-block"
            disabled={true}
            type="submit"
          >
            <Glyphicon
              bsClass="glyphicon"
              className="spinning"
              glyph="refresh"
            >
              <span
                className="spinning glyphicon glyphicon-refresh"
              />
            </Glyphicon>
            Creating…
          </button>
        </Button>
      </LoaderButton>
    </form>
  </div>
</NewNote>
`;

exports[`New Note component renders when submit with file bigger than MAX_ATTACHMENT_SIZE 1`] = `
<NewNote
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="NewNote"
  >
    <form
      onSubmit={[Function]}
    >
      <FormGroup
        bsClass="form-group"
        controlId="content"
      >
        <div
          className="form-group"
        >
          <FormControl
            bsClass="form-control"
            componentClass="textarea"
            onChange={[Function]}
            value=""
          >
            <textarea
              className="form-control"
              id="content"
              onChange={[Function]}
              value=""
            />
          </FormControl>
        </div>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        controlId="file"
      >
        <div
          className="form-group"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="file"
            >
              Attachment
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="file"
          >
            <input
              className=""
              id="file"
              onChange={[Function]}
              type="file"
            />
          </FormControl>
        </div>
      </FormGroup>
      <LoaderButton
        block={true}
        bsSize="large"
        bsStyle="primary"
        disabled={true}
        isLoading={false}
        loadingText="Creating…"
        text="Create"
        type="submit"
      >
        <Button
          active={false}
          block={true}
          bsClass="btn"
          bsSize="large"
          bsStyle="primary"
          className="LoaderButton "
          disabled={true}
          type="submit"
        >
          <button
            className="LoaderButton  btn btn-lg btn-primary btn-block"
            disabled={true}
            type="submit"
          >
            Create
          </button>
        </Button>
      </LoaderButton>
    </form>
  </div>
</NewNote>
`;

exports[`New Note component renders without crashing 1`] = `
<div
  className="NewNote"
>
  <form
    onSubmit={[Function]}
  >
    <FormGroup
      bsClass="form-group"
      controlId="content"
    >
      <FormControl
        bsClass="form-control"
        componentClass="textarea"
        onChange={[Function]}
        value=""
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      controlId="file"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Attachment
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        type="file"
      />
    </FormGroup>
    <LoaderButton
      block={true}
      bsSize="large"
      bsStyle="primary"
      disabled={true}
      isLoading={false}
      loadingText="Creating…"
      text="Create"
      type="submit"
    />
  </form>
</div>
`;
